sdk/core/Azure.Core/README.md

```diff
@@ -1,45 +1,100 @@
+so that once you learn how to use these APIs in one client library, you will know how to use them in other client libraries.
 
-# Installing
+The main shared concepts of Azure.Core (and so all Azure SDK libraries using Azure.Core) include:
+
+1. Unified APIs for configuring service clients, e.g. configuring retries, logging.
+2. Unified APIs for accessing HTTP responses.
+3. Unified APIs for consuming long running operations (LROs).
+4. Unified APIs for consuming asynchronous streams (```IAsyncEnumerable<T>```) by paging.
+5. Unified exception hierarchy for reporting errors from service requests.
+6. Abstractions for representing Azure SDK credentials.
+
+The following sections will explain these shared concepts in more detail.
 
-Nuget package Azure.Core avaliable on https://www.nuget.org/packages/Azure.Core/
+# Usage Scenarios and Samples
 
-# Hello World
+## Configuring Service Clients Using ```ClientOptions```
+Azure SDK client libraries typically expose one or more _service client_ types that 
+are the main starting points for calling corresponding Azure services. 
+You can easily find these client types as their names end with the word _Client_. 
+For example, ```BlockBlobClient``` can be used to call blob storage service, 
+and ```KeyClient``` can be used to access KeyVault service cryptographic keys. 
 
-```c#
-public async Task HelloWorld()
+These client types can be instantiated by calling a simple constructor, 
+or its overload that takes various configuration options. 
+These options are passed as a parameter that extends ```ClientOptions``` class exposed by Azure.Core.
+Various service specific options are usually added to its subclasses, but a set of SDK-wide options are 
+available directly on ```ClientOptions```.
+
+```csharp
+public void ConfigureServiceClient()
 {
-    // create http pipeline
-    var options = new HttpPipeline.Options();
-    HttpPipeline pipeline = HttpPipeline.Create(options, sdkName: "test", sdkVersion: "1.0");
-
-    // create http message
-    using (HttpMessage message = pipeline.CreateMessage(options, cancellation: default)) {
-
-        // set message URI
-        var uri = new Uri(@"https://raw.githubusercontent.com/Azure/azure-sdk-for-net/master/README.md");
-        message.SetRequestLine(HttpVerb.Get, uri);
-
-        // add headers
-        message.AddHeader("Host", uri.Host);
-
-        // send message
-        await pipeline.SendMessageAsync(message).ConfigureAwait(false);
-
-        // process response
-        Response response = message.Response;
-        if (response.Status == 200) {
-            var reader = new StreamReader(response.ContentStream);
-            string responseText = reader.ReadToEnd();
-        }
-        else throw new RequestFailedException(response);
-    }       
+    // BlobConnectionOptions inherits/extends ClientOptions
+    ClientOptions options = new BlobConnectionOptions();     
+    
```

Comment:
could we include an example of calling the simple constructor? So that we show people both options here?

Reply:
I added 
```c#
    // if you don't specify the options, default options will be used, e.g.
    var clientWithDefaultOptions = new BlobContainerClient(connectionString, "container");
```